### PR TITLE
tests, integration-tests: port search tests to spread

### PR DESCRIPTION
**Pull request by fgimenez**

### CODE REVIEW

File: tests/searching/task.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  .*?\
+  canonical-pc-linux +.*? *\n\
+  .*?\
+  go-example-webserver +.*? *\n\
+  .*?\
+  hello-world +.*? *\n\
+  .*?\
+  ubuntu-clock-app +.*? *\n\
+  .*?\
+  ubuntu-core +.*? *\n\
```
Comment (niemeyer) [inline tests/searching/task.yaml]: This is a test that will most likely be broken very soon, as we won't want to list everything.

For now, can we please keep just hello-world on the list, which is something we depend on for other tests anyway, so we reduce the changes of this test being broken on other unrelated changes?

Comment (niemeyer) [inline tests/searching/task.yaml]: Probably okay to keep ubuntu-clock-app there as well for the time being, at least to the ordering and the fact we're showing more than a single snap. But let's please try to constraint here and in other tests the number of snaps we assume exists on the store.

File: tests/searching/task.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  .*"
+  actual=$(snap find)
+  echo "$actual" | grep -Pzq "$expected" || exit 1
+
+  echo "Exact matches"
+  for snapName in hello-world http ubuntu-clock-app go-example-webserver
```
Comment (niemeyer) [inline tests/searching/task.yaml]: Along the same lines, can we please do the test with hello-world and ubuntu-clock-app only?

**Comment by niemeyer** [comment]
A couple of suggestions and LGTM!

**Comment by fgimenez** [comment]
@niemeyer thanks for your review! comments addressed

**Comment by fgimenez** [comment]
retest this please

**Comment by mvo5** [comment]
:+1:
